### DISCUSSION TITLE
Write parent node or mutliple nodes at once

### DISCUSSION THREAD

**Discussion by ghost** [eclipse-milo/milo#1142] [General]
I want to write multiple nodes at once. Using org.eclipse.milo.opcua.sdk.client.api.services.AttributeServices#writeValues is basically a call to singe writeValue if Im correct.

Is there a way to just write a parent node and all child node of it at once, or am I missing something completely here?

Thanks for your help :-)

**Comment by kevinherron** [comment]
That method takes a List of `NodeId` and a corresponding List of `DataValue`. It can write to multiple Nodes.

**Reply by kevinherron** [reply]
You can also use `OpcUaClient::write`, which is a little "lower level" in that its parameters better reflect the actual definition of the Write service from OPC UA Part 4.

https://github.com/eclipse/milo/blob/master/opc-ua-sdk/sdk-client/src/main/java/org/eclipse/milo/opcua/sdk/client/OpcUaClient.java#L526

**Reply by ghost** [reply]
The values in that list are then written in one write process to the plc? And not one after each other?

**Reply by kevinherron** [reply]
They are sent in one request to the server. What happens on the server is implementation specific. 

For a PLC it's reasonable to assume they are written during the same "scan" or similar process, though.

**Reply by ghost** [reply]
Thats all I need. Thanks.
